packages/peregrine/lib/talons/CartPage/GiftOptions/useGiftOptions.js

```diff
@@ -0,0 +1,206 @@
+                    extension_attributes: {
+                        wrappingAddPrintedCard:
+                            newGiftOptions.include_printed_card,
+                        wrappingAllowGiftReceipt:
+                            newGiftOptions.include_gift_receipt
+                    }
+                }
+            };
+
+            return sendRequest({
+                options: {
+                    method: 'POST',
+                    body: JSON.stringify(data)
+                }
+            });
+        },
+        [sendRequest]
+    );
+    /**
+     * End of Temp stuff.
+     */
+
+    const updateGiftOptions = useCallback(
+        optionsToUpdate => {
+            const newGiftOptions = {
+                cart_id: cartId,
+                include_gift_receipt: includeGiftReceipt,
+                include_printed_card: includePrintedCard,
+                gift_message: giftMessage,
+                ...optionsToUpdate
+            };
+            setGiftOptions({
+                variables: newGiftOptions
+            });
+            /**
+             * TODO remove this after GQL gets gift options coverage.
+             */
+            updateUsingRest(newGiftOptions);
+        },
+        [
+            updateUsingRest,
+            setGiftOptions,
+            cartId,
+            includeGiftReceipt,
+            includePrintedCard,
+            giftMessage
+        ]
+    );
+
+    /**
+     * Throttling message update. Only make 1 mutation
+     * every 10 seconds. This is to save on bandwidth.
+     *
+     * More info: https://lodash.com/docs/4.17.15#throttle
+     */
+    const throttledMessageUpdate = useMemo(() => {
+        return throttle(
+            (updateGiftOptions, newGiftMessage) => {
+                updateGiftOptions({
+                    gift_message: newGiftMessage
+                });
+            },
+            10000,
+            {
+                leading: false
+            }
+        );
+    }, []);
+
+    const updateGiftMessage = useCallback(
+        newGiftMessage => {
+            setGiftMessage(newGiftMessage);
+            throttledMessageUpdate(updateGiftOptions, newGiftMessage);
+        },
+        [setGiftMessage, throttledMessageUpdate, updateGiftOptions]
+    );
+
+    const toggleIncludeGiftReceiptFlag = useCallback(() => {
+        setIncludeGiftReceipt(!includeGiftReceipt);
+        updateGiftOptions({
+            include_gift_receipt: !includeGiftReceipt
+        });
+    }, [updateGiftOptions, includeGiftReceipt, setIncludeGiftReceipt]);
+
+    const toggleIncludePrintedCardFlag = useCallback(() => {
+        setIncludePrintedCard(!includePrintedCard);
+        updateGiftOptions({
+            include_printed_card: !includePrintedCard
+        });
+    }, [updateGiftOptions, includePrintedCard, setIncludePrintedCard]);
+
+    /**
+     * Fetch gift options for a given cart id.
+     */
+    const { data } = useQuery(GET_GIFT_OPTIONS_QUERY, {
```

Comment:
Use a lazy query like in the other components within `CartPage`. We can only query for data if we have a cartId, and we don't always have a cartId (at least not right now).